src/components/SignUpForm/index.test.jsx

```diff
@@ -0,0 +1,415 @@
+import React from "react";
+import { render, screen, fireEvent, waitFor } from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+
+import SignUpForm from "./index";
+
+// TODO: refactor the code and extract repetitive code to helper functions
```

Comment:
that comment wasn't for a separate PR. like I said we don't commit incomplete code because there is high probability no one will fix this PR

fix and identify the common functions, the rest file should not be 400 lines